### ISSUE TITLE
Database getting corrupted

### ISSUE THREAD

**Issue by MCUniversity** [TomPrograms/stormdb#36]
I found 2 bugs that I don't understand how are happening and have been trying to catch for a while now, and while working on a project I finally caught one of them.
I am making a discord bot, and I am using stormdb.
I ran a command that updated the database and it was supposed to change to this:
![image](https://user-images.githubusercontent.com/42188518/145890771-291dc271-6f1a-42ab-abd2-96cd00c41d6c.png)
But instead it changed to this:
![image](https://user-images.githubusercontent.com/42188518/145890670-4f050194-1a6d-4053-844d-75cb511d1623.png)

This happened when changing `config.channel.log.638385234915885076` from an object to a string.
I saved an entire object by accident to the database instead of just the ID I wanted.
After using db.set like this:
`db.set('config.channels.log.${interaction.guild.id}', logs.id).save()` (replace the ' with `)
and changing it from saving the entire object to just the ID, the garbage just appeared at the end of the JSON file.
You can actually still see remains of that unwanted object I saved by accident in the garbage. Its an object that represents a  discord channel and you can see the keys from the object: threads, permissionOverwrites, topic, rateLimitPerUser, etc.
No errors appeared.

**Comment by MCUniversity** [comment]
The 2nd bug I found happens randomly and I was not able to reproduce it, but sometimes random keys and all values inside them just disappear without any explanation.
I would just out of nowhere start getting the `Cannot run .value() on undefined` error randomly, and look at the database to find random keys and all data inside them from the root of the JSON file missing.

**Comment by TomPrograms** [comment]
Sorry for the long delay in writing a reply to this. I wonder if the problems are occurring because you are using `.save()` too rapidly. You only need to use `.save()` when you want to write a copy of the database to the disk, e.g. before the program powers down, not after every operation.

**Comment by MCUniversity** [comment]
The program is never supposed to power down, it runs on a server continuously. The server and program can crash tho, and it has happened before so I use .save() whenever I write something to the DB to make sure I lose no data.

A fix would then be periodically running .save() every X amount of time forever?

Closing this since I no longer need a solution as I moved to using mongodb for my database.
